test/e2e/config/e2e_conf.yaml

```diff
@@ -110,8 +110,8 @@ variables:
   KUBERNETES_VERSION: "v1.27.1"
   # INIT_WITH_KUBERNETES_VERSION will be used here
   # https://github.com/kubernetes-sigs/cluster-api/blob/bb377163f141d69b7a61479756ee96891f6670bd/test/e2e/clusterctl_upgrade.go#L170
-  # INIT_WITH_KUBERNETES_VERSION  used by management cluster upgrade and it is fixed to 1.23.8 because v1a4 capi does not support k8s version higher
-  INIT_WITH_KUBERNETES_VERSION: "v1.23.8"
+  # INIT_WITH_KUBERNETES_VERSION  used by management cluster upgrade
+  INIT_WITH_KUBERNETES_VERSION: "${INIT_WITH_KUBERNETES_VERSION}"
```

Comment:
Seems unnecessary to template this. Will it not anyway be overridden by the environment variables? No big deal though

Reply:
that was my expectation but in the test https://jenkins.nordix.org/job/metal3_capm3_from-release-0-5_main_e2e_clusterctl_upgrade_test_ubuntu/3/consoleFull
Seems like capi test did not reach the env var so I thought config file will solve this

Reply:
Hmm may be worth investigating why that is the case. But I'm happy to take this in to get green CI first :slightly_smiling_face: